app/lib/domains.server.ts

```diff
@@ -0,0 +1,146 @@
+import * as dns from './dns.server';
+import { RecordStatus } from '@prisma/client';
+import { prisma } from '~/db.server';
+import type { RecordType } from '@prisma/client';
+
+interface DnsRecord {
```

Comment:
Throughout this file, you use `Record & DnsRecord`.  I wonder if you should declare a type and use it, then export?

```ts
export type Domain = Record & DnsRecord;
```

Then all your functions below can take `(domain: Domain)`